config/application.rb

```diff
@@ -56,5 +57,7 @@ class Application < Rails::Application
     config.autoload_paths << Rails.root.join('lib')
 
     config.active_job.queue_adapter = :sidekiq
+
+    config.middleware.insert_before 0, SameSiteSecurity::Middleware
```

Comment:
Perhaps `::SameSiteSecurityMiddleware` ? Either way, the class/module/file layout isn't correct. The convention is that nested classes are in directories, so `SameSiteSecurity::Middleware` should be `lib/same_site_security/middleware.rb`